src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3214,10 +3215,32 @@ internal static BuildIntegratedInstallationContext CreateInstallationContextForP
                 .Except(unsuccessfulFrameworks)
                 .ToList();
 
+            var areAllPackagesConditional = DoesPackageAppearWithDifferentVersions(packageIdentityId, originalPackageSpec);
+
             return new BuildIntegratedInstallationContext(
                 successfulFrameworksWithPackage,
                 unsuccessfulFrameworks.Union(frameworksWithoutResultingPackage).Distinct(),
-                originalFrameworks);
+                originalFrameworks,
+                areAllPackagesConditional);
+        }
+
+        private static bool DoesPackageAppearWithDifferentVersions(string packageIdentityId, PackageSpec packageSpec)
+        {
+            HashSet<VersionRange> versions = default;
+
+            foreach (var framework in packageSpec.TargetFrameworks)
+            {
+                foreach (var dependency in framework.Dependencies)
+                {
+                    if (dependency.Name == packageIdentityId)
+                    {
+                        versions ??= new();
+                        versions.Add(dependency.LibraryRange.VersionRange);
+                        break;
```

Comment:
I guess we could return from here if Versions?.Count > 1`

Reply:
We certainly could, but this isn't a hot path, and projects with 3 frameworks aren't that common, so I didn't bother optimizing.